dask/dataframe/_hyperloglog.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*
+u"""Implementation of HyperLogLog
+
+This implements the HyperLogLog algorithm for cardinality estimation, found
+in
+
+    Philippe Flajolet, Éric Fusy, Olivier Gandouet and Frédéric Meunier.
+        "HyperLogLog: the analysis of a near-optimal cardinality estimation
+        algorithm". 2007 Conference on Analysis of Algorithms. Nice, France
+        (2007)
+
+"""
+from __future__ import division
+
+import numpy as np
+import pandas as pd
+
+def compute_first_bit(a):
+    # Consider making this less memory-hungry
+    #bits = np.bitwise_and.outer(a, 1<<np.arange(31, -1, -1))
+    bits = np.bitwise_and.outer(a, 1<<np.arange(32))
+    bits = bits.cumsum(axis=1).astype(np.bool)
+    return 33 - bits.sum(axis=1)
+
+def terrible_dataframe_hash(df):
```

Comment:
My understanding (which is pretty light and built in a rush) is that one plan is to start the Python processes with the same PYTHONHASHSEED environment variable in the different worker processes.  Generally ensuring uniform environment across a cluster is tricky, especially when we try to support multiple platforms and the like.  People like to deploy distributed systems in a variety of ways so this would be hard to guarantee.  I'm really hesitant to say that dask.distributed only works if you set a consistent hash seed when starting it up.  I'm way more willing to say that HyperLogLog only works on common dtypes.

However, it's quite likely that I do not understand the issue fully.  I'm blowing through some of these issues rather quickly in an effort to be caught up.  If I've misunderstood the requirements here then please let me know and I'll walk through this more carefully.